src/Functions/array/arrayIntersect.cpp

```diff
@@ -549,44 +550,101 @@ ColumnPtr FunctionArrayIntersect::execute(const UnpackedArrays & arrays, Mutable
                 }
             }
 
-            prev_off[arg_num] = off;
-            if (arg.is_const)
-                prev_off[arg_num] = 0;
-
+            if (arg_num)
```

Comment:
The comment is needed here.
We update offsets for all the arrays except the first one. Offsets for the first array would be updated later. It is needed to iterate the first array again so that the elements in the result would have fixed order.